### PR TITLE
Gazebo simulation tools.

### PR DESCRIPTION
**Pull request by bperseghetti**
**Problem solved by this pull request**
This PR allows for a clean running of gazebo simulation for sitl and hitl with gz_sim.sh and advanced runtime generation of worlds and models. This is used in conjunction with [PR 651](https://github.com/PX4/PX4-SITL_gazebo/pull/651) in PX4-SITL_gazebo.

**Test data / coverage**
Tested extensively.

### CODE REVIEW

File: ROMFS/px4fmu_common/init.d-posix/airframes/1064_cupcar [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+	param set CBRK_AIRSPD_CHK 162128
+
+	param set GND_MAX_ANG 0.6
+	param set GND_WHEEL_BASE .15
+
+fi
+
+set MAV_TYPE 10
+
+set MIXER_FILE etc/mixers-sitl/rover_ackermann_sitl.main.mix
```
Comment (Jaeyoung-Lim) [inline ROMFS/px4fmu_common/init.d-posix/airframes/1064_cupcar:37]: Could you separate the addition of a new SITL model to a separate PR? The changes seem quite orthogonal

File: Tools/gz_sim.sh [hunk 2]
```diff
@@ -0,0 +1,191 @@
+#!/bin/bash
+# run multiple instances of the 'px4' binary, with the gazebo SITL simulation
+# It assumes px4 is already built, with 'make px4_sitl_default gazebo'
+
+# The simulator is expected to send to TCP port 4560+i for i in [0, N-1]
+# For example gazebo can be run like this:
+#./Tools/gazebo_sitl_multiple_run.sh -n 10 -m iris
```
Comment (Jaeyoung-Lim) [inline Tools/gz_sim.sh:7]: This seems like a copy paste coming over from another script

**Review by Jaeyoung-Lim — CHANGES_REQUESTED** [review]
Thanks for the contribution, I have a few questions regarding this PR

- I am a bit concerned that this introduces another way of running SITL which is not part of the CI. How do you see this fit in with other simulations and the already existing `sitl_run.sh`?
- gz_sim.sh seems to include a duplicate functionality of `gazebo_sitl_multiple_run.sh` are you trying to replace this script?
- the `reboot_mavlink_shell.py` seems like a duplicate of boot_now.py, are you trying to replace this script?

Additional:
- The NXP rover model being added to SITL seems orthogonal to the other tool changes, therefore I think it should be a separate PR

File: ROMFS/px4fmu_common/init.d-posix/airframes/1064_cupcar (hunk 1 above)
Comment (bperseghetti) [reply ROMFS/px4fmu_common/init.d-posix/airframes/1064_cupcar:37]: I wanted to simplify and tie together the changes also in PX4-SITL_gazebo for enabling the cupcar by including it in this PR as it was added to the generator list.

File: Tools/gz_sim.sh (hunk 2 above)
Comment (bperseghetti) [reply Tools/gz_sim.sh:7]: I think if you were to look at it more closely you would see that it's a superset of all potential uses/runs for gazebo, hence why parts might look familiar. However, unlike the others, it allows for HITL and SITL, gen from json for both worlds and models (SITL and HITL), setting of simulation threads with smart auto-limits, and much more. Currently there is nothing that provides even a fraction of this functionality.

Comment (Jaeyoung-Lim) [reply Tools/gz_sim.sh:7]: @bperseghetti I didn't mean that you copy pasted the whole script... The comment says the script can be run with `./Tools/gazebo_sitl_multiple_run.sh -n 10 -m iris`

Comment (bperseghetti) [reply Tools/gz_sim.sh:7]: Unless you are specifically only talking about the comments, in which case they could be expanded more, but I think it would be best in the form of a dev guide update, the comment section of this otherwise would be very significant. Since this will be replacing the multiple_sitl_gazebo.sh I wanted to let people know that the same functionality is still preserved.

Comment (Jaeyoung-Lim) [reply Tools/gz_sim.sh:7]: @bperseghetti Yes, I was specificly talking about the comments - The comments are not talking about the new script, but explaining how to run the old script.

Comment (bperseghetti) [reply Tools/gz_sim.sh]: Ok, thanks for clarification, I have updated the comments.

Comment (Jaeyoung-Lim) [reply Tools/gz_sim.sh]: Thanks

**Comment by bperseghetti** [comment]
Think this has all the requested changes and should be all good to go.

**Comment by Jaeyoung-Lim** [comment]
@bperseghetti Can you clarify on the following points I mentioned above?
> - gz_sim.sh seems to include a duplicate functionality of gazebo_sitl_multiple_run.sh are you trying to replace this script?
> - the reboot_mavlink_shell.py seems like a duplicate of boot_now.py, are you trying to replace this script?

**Review by Jaeyoung-Lim — CHANGES_REQUESTED** [review]
Thanks! Looks good to me on my side!

- Please squash and clean up the commits
- There is a conflict with `gazebo_sitl_multiple_run.sh` precenting this PR from being merged, please rebase

**Comment by LorenzMeier** [comment]
What is the next step here?
